.github/workflows/test.yaml

```diff
@@ -0,0 +1,41 @@
+name: CI Tests
+on:
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+  push:
+    branches:
+      - 'master'
+    paths-ignore:
+      - 'README.md'
+
+permissions:
+  contents: read
+
+jobs:
+  go-fmt-and-vet:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+    - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+      with:
+        go-version: '1.20'
```

Comment:
Not a more recent version?

Reply:
I just copied and pasted [raft's](https://github.com/hashicorp/raft/blob/main/.github/workflows/ci.yml) as a basis assuming they're the best maintained shared library. :grimacing:  We can bump it. I'm not really sure what the optimal strategy is for a little library like this with no deps and no-post-1.14 features afaik.